network/src/protocols/health_checker/mod.rs

```diff
@@ -76,6 +76,37 @@ pub struct HealthCheckerNetworkSender {
     inner: NetworkSender<HealthCheckerMsg>,
 }
 
+#[async_trait]
+impl crate::application::interface::ApplicationPeerNetworkIdSender<HealthCheckerMsg>
```

Comment:
nit: any reason we aren't just importing the trait?

Reply:
In me removing the `SingleNetworkSender`, since there are actually 2 traits applied on this sender, the usage of `send_rpc` is for the `PeerId` usage, but needs to be `PeerNetworkId` to be in the `NetworkInterface`.

This is the only application that uses `PeerId`, because it's tied to a single network.  I figured at a later time we could cleanup the `HealthChecker` to be combined for multiple networks.